docs/custom-flows/manage-totp-mfa.mdx

```diff
@@ -0,0 +1,353 @@
+---
+
+# Build a custom flow for managing TOTP-based multi-factor authentication
+
+> [!WARNING]
+> This guide is for users who want to build a *custom* user interface using the Clerk API. To use a *prebuilt* UI, you should use Clerk's [Account Portal pages](/docs/account-portal/overview) or [prebuilt components](/docs/components/overview).
+
+[Multi-factor verification (MFA)](/docs/authentication/configuration/sign-up-sign-in-options) is an added layer of security that requires users to provide a second verification factor to access an account.
+
+One of the options that Clerk supports for MFA is **Authenticator applications (also known as TOTP - Time-based One-time Password)**. This guide will walk you through how to build a custom flow that allows users to manage their TOTP settings.
+
+> [!TIP]
+> To learn how to build a custom flow for managing SMS MFA, see the [dedicated guide](/docs/custom-flows/manage-sms-mfa).
+
+<Steps>
+
+### Enable multi-factor authentication
+
+For your users to be able to enable MFA for their account, you need to enable MFA as an MFA authentication strategy in your Clerk application.
+
+1. Navigate to the [Clerk Dashboard.](https://dashboard.clerk.com/last-active?path=user-authentication/multi-factor)
+2. In the navigation sidebar, select **User & Authentication > Multi-factor**.
+3. Enable **Authenticator application** and **Backup codes** and select **Save**.
+
+### Create the multi-factor management flow
+
+This example is written for Next.js App Router but it can be adapted for any React meta framework, such as Remix or Gatsby.
+
+This example consists of two pages:
+- The main page where users can manage their MFA settings
+- The page where users can add TOTP MFA.
+
+Use the following tabs to view the code necessary for each page.
+
+<Tabs items={["Manage MFA page", "Add TOTP page"]}>
+  <Tab>
+    ```jsx {{ filename: '/app/account/manage-mfa/page.tsx' }}
+    'use client';
+
+    import * as React from 'react';
+    import { useUser } from '@clerk/nextjs';
+    import Link from 'next/link';
+    import { BackupCodeResource } from '@clerk/types';
+
+    // If TOTP is enabled, provide the option to disable it
+    const TotpEnabled = () => {
+      const { user } = useUser();
+
+      const disableTOTP = async () => {
+        await user?.disableTOTP();
+      };
+
+      return (
+        <div>
+          <p>
+            TOTP via authenication app enabled -{' '}
+            <button onClick={() => disableTOTP()}>Remove</button>
+          </p>
+        </div>
+      );
+    };
+
+    // If TOTP is disabled, provide the option to enable it
+    const TotpDisabled = () => {
+      return (
+        <div>
+          <p>
+            Add TOTP via authentication app -{' '}
+            <Link href="/account/manage-mfa/add">
+              <button>Add</button>
+            </Link>
+          </p>
+        </div>
+      );
+    };
+
+    // Generate and display backup codes
+    function GenerateBackupCodes() {
```

Comment:
In the original code, I had this as some basic logic in the `ManageTOTP` (now `ManageMFA`). Now that this is is own component, its a near duplicate to the `DisplayBackupCodes` component from the `/app/account/manage-mfa/add/page.tsx` section.

Let's either revert this to the more basic version from the original code, or use one component (this one probably) and import it on the other page.